### DISCUSSION TITLE
"Sign off and commit suggestion" should let users choose email address for commit and signoff

### DISCUSSION THREAD

**Discussion by cornelius** [community/community#68579] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

The "Sign off and commit suggestion" feature let's you create commits from suggestions in pull requests comments including a "Signed-off-by" trailer in the commit message. At the moment the dialog box which opens when you click the "Sign off and commit suggestion" button states which email address is used for the signoff. There is no way to choose the email address when the user has more than one email address. It always uses the primary email address. This is a problem when a user for example commits to different repositories in different roles (e.g. personal work and work on behalf on an employer). The user would have to change the global primary email address in their account each time before using the "Sign off and commit suggestion" button.

It would be much better, if there would be a selection box, where the user can choose which email address is used for the signoff. This would be in line with all the other places where the user can choose the email address used for web-based commits.

**Comment by funnelfiasco** [comment]
This also presents a problem for some [DCO enforcement apps](https://probot.github.io/apps/dco/). I had a check fail because it expected my work email address (which I had used for the original commit) but the GitHub web interface used my personal email address and apparently the bot cares about that.
